pl_examples/full_examples/imagenet/dali_utils.py

```diff
@@ -0,0 +1,84 @@
+import urllib.request
+import zipfile
+import os
+
+from nvidia.dali.pipeline import Pipeline
+import nvidia.dali.ops as ops
+import nvidia.dali.types as types
+
+url = 'https://s3.amazonaws.com/pytorch-tutorial-assets/imagenet_1k.zip'
+filename = './imagenet_1k.zip'
+dataset_root = './imagenet_1k'
+
+
+def prepare_imagenet_1k():
+    if not os.path.exists(dataset_root):
+        if not os.path.exists(filename):
+            urllib.request.urlretrieve(url, filename)
+
+        with zipfile.ZipFile(filename, 'r') as zip_ref:
+            zip_ref.extractall(".")
+
+
+# modified from NVIDIA/DALI/docs/examples/pytorch/resnet50/main.py
+class HybridTrainPipe(Pipeline):
+    def __init__(self, batch_size, num_threads, local_rank, world_size, data_dir, crop=224, dali_cpu=False):
+        super(HybridTrainPipe, self).__init__(batch_size=batch_size, num_threads=num_threads, device_id=local_rank,
+                                              seed=12 + local_rank)
+        self.input = ops.FileReader(file_root=data_dir, shard_id=local_rank, num_shards=world_size, random_shuffle=True)
+        # let user decide which pipeline works him bets for RN version he runs
+        dali_device = 'cpu' if dali_cpu else 'gpu'
+        decoder_device = 'cpu' if dali_cpu else 'mixed'
+        # This padding sets the size of the internal nvJPEG buffers
+        # to be able to handle all images from full-sized ImageNet
+        # without additional re-allocations
+        device_memory_padding = 211025920 if decoder_device == 'mixed' else 0
```

Comment:
how is the exact number derived?

Reply:
This number comes up in the dali repo with no explanation. 
I suppose this might the max file size in the imagenet data?